### PR TITLE
Expose CPack compression controls

### LINKED ISSUE #83: Expose CPack archive compression controls
## Summary

Expose or formally document wrapper support for modern CPack compression settings, including:

- `CPACK_COMPRESSION_LEVEL`
- `CPACK_ARCHIVE_COMPRESSION_LEVEL`
- `CPACK_DEBIAN_COMPRESSION_LEVEL`

Compression selection affects output size, packaging time, and reproducibility and should not require undocumented variable injection.

## Acceptance criteria

- Define wrapper arguments or a documented pass-through contract.
- Validate levels and generator compatibility where CPack provides deterministic rules.
- Preserve existing defaults when omitted.
- Add tests for a supported archive generator and Debian packaging when practical.
- Document version gates and precedence between generic and generator-specific settings.

## Edge cases

Cover unsupported algorithms or levels, conflicting generic and generator-specific values, and older CMake versions.

### PR DESCRIPTION
**Pull request by jkammerland**
Adds generic, archive, and Debian CPack compression controls with generator-aware validation and pass-through override support.

Validation: CMake 4.3/4.4 suites, real archive and Debian packages, platform selector checks, and the old-version guard.

Closes #83

### CODE REVIEW

**Comment by jkammerland** [comment]
@codex review

File: export_cpack.cmake [hunk 1]
```diff
@@ -876,6 +898,10 @@ function(_execute_deferred_cpack_config)
       DEFAULT_COMPONENTS
       ENABLE_COMPONENT_INSTALL
       ARCHIVE_FORMAT
+      COMPRESSION_LEVEL
+      ARCHIVE_COMPRESSION_LEVEL
+      DEBIAN_COMPRESSION_TYPE
+      DEBIAN_COMPRESSION_LEVEL
```
[the remark this answers is not in the document]
Comment (jkammerland) [reply export_cpack.cmake]: Fixed in 1df6404. Additional variable values are opaque, with a regression proof.

**Comment by jkammerland** [comment]
@codex review

**Comment by jkammerland** [comment]
@codex review

File: export_cpack.cmake [hunk 2]
```diff
@@ -1138,6 +1193,145 @@ function(_execute_deferred_cpack_config)
...
+      ZIP_ZSTD
+      CYGWINBINARY
+      CYGWINSOURCE
+      FREEBSD)
+  set(_tip_archive_zstd_generators 7Z_ZSTD TZST)
+  set(_tip_has_binary_archive_generator FALSE)
+  foreach(_tip_generator_upper IN LISTS _tip_generators_upper)
+    if(_tip_generator_upper IN_LIST _tip_archive_generators)
+      set(_tip_has_binary_archive_generator TRUE)
+    endif()
```
[the remark this answers is not in the document]
Comment (jkammerland) [reply export_cpack.cmake]: Fixed in 40fae28. Compression validation now uses the final CPACK_GENERATOR override, with both valid and invalid override proofs.

File: export_cpack.cmake [hunk 3]
```diff
@@ -1138,6 +1193,141 @@ function(_execute_deferred_cpack_config)
...
+    set(_tip_debian_max_compression_level 19)
+  else()
+    set(_tip_debian_max_compression_level 9)
+  endif()
+
+  if(_tip_archive_compression_level_explicit)
+    if(NOT _tip_has_binary_archive_generator)
+      project_log(FATAL_ERROR "ARCHIVE_COMPRESSION_LEVEL requires an archive generator")
+    endif()
+    _tip_validate_cpack_compression_level(ARCHIVE_COMPRESSION_LEVEL "${ARG_ARCHIVE_COMPRESSION_LEVEL}" "${_tip_archive_max_compression_level}")
```
[the remark this answers is not in the document]
Comment (jkammerland) [reply export_cpack.cmake]: Fixed in 40fae28. Same-variable pass-through overrides now bypass discarded wrapper validation, with regression coverage.

**Comment by jkammerland** [comment]
@codex review

**Comment by jkammerland** [comment]
@codex review

**Comment by jkammerland** [comment]
@codex review

**Comment by jkammerland** [comment]
@codex review

File: CPack-Tutorial.md [hunk 4]
```diff
@@ -354,6 +354,40 @@ export_cpack(
...
+
+The wrapper rejects levels outside the range shared by every affected binary and source generator. CPack writes generic and archive-specific compression variables to both `CPackConfig.cmake` and `CPackSourceConfig.cmake`; the default compressed source generators therefore usually constrain these settings to `0` through `9`. `TAR` and `TZ` do not consume a configurable compression level and therefore do not constrain the range, while `7Z_STORE` and `ZIP_STORE` retain CPack's `0` through `9` limit. Levels `10` through `19` are accepted only when every compression consumer is deterministically Zstandard-based, or when a generator-specific level overrides the generic level for the non-Zstandard consumers. To use a high Zstandard level for both configs, select a Zstandard source generator explicitly:
```
Comment (jkammerland) [inline CPack-Tutorial.md]: Too verbose. Keep it short.

Comment (jkammerland) [reply CPack-Tutorial.md]: Fixed in 6e15d2a. Kept only supported values, precedence, and version requirements.

**Comment by jkammerland** [comment]
@codex review

File: export_cpack.cmake [hunk 5]
```diff
@@ -197,6 +206,35 @@ function(_tip_finalize_registered_exports_for_cpack)
   endforeach()
 endfunction()
 
+function(_tip_validate_cpack_compression_level argument_name value max_level)
```
[the remark this answers is not in the document]
Comment (jkammerland) [reply export_cpack.cmake]: Fixed in 136f402. All built-in commands added by the compression implementation are uppercase. CMake 4.3.1 passed 42/42 tests.

**Comment by jkammerland** [comment]
@codex review
